packages/opentelemetry-metrics/src/export/aggregators/histogram.ts

```diff
@@ -0,0 +1,96 @@
+/*!
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Aggregator, Point } from '../types';
+import { HrTime } from '@opentelemetry/api';
+import { hrTime } from '@opentelemetry/core';
+
+export type Checkpoint = {
+  buckets: {
+    boundaries: number[];
+    counts: number[];
+  };
+  sum: number;
+  count: number;
+};
+
+/** Basic aggregator which calculates a Sum from individual measurements. */
+export class HistogramAggregator implements Aggregator {
```

Comment:
Seems like there is some shared things in these like the `toPoint` method. Do you think there's value in making all these aggregators share a base class?

Reply:
Hard to tell. I don't know where does the `toPoint` method comes from because the spec expect a method called `checkpoint` (like i've done here) : 
https://github.com/jmacd/opentelemetry-specification/blob/jmacd/draft_metric_sdk_spec/specification/sdk-metric.md#L25
The go implementation seems to be right:
https://github.com/open-telemetry/opentelemetry-go/blob/6489b07bf55620f2d66e8da64c00d76bd3f1cb75/sdk/export/metric/metric.go#L140

Reply:
As said on a comment from @mayurkale22, i think we need to change our `Aggregator` interface that doesnt match the spec right now and that case its better to make a different PR for that